discovery/db_discovery.go

```diff
@@ -240,7 +240,7 @@ func (dbo *DBOrchestratorPoolCache) cacheDBOrchs() error {
 		return fmt.Errorf("could not retrieve orchestrators from DB: %v", err)
 	}
 
-	resc, errc := make(chan *common.DBOrch), make(chan error)
+	resc, errc := make(chan *common.DBOrch, len(orchs)), make(chan error, len(orchs))
```

Comment:
So I think there is actually a bug with the timeout behavior because on L284 when a timeout is triggered the break statement will break out of the select statement and not the for loop. As a result, when the timeout is triggered control flow will just move onto the next iteration of the for loop. If that is the case, then I don't think there was a memory leak since the the for loop would run N times once for each of the N goroutines created for an O. 

That being said, if we fix the timeout behavior then there would be a memory leak without this change so it might make sense to fix the timeout behavior alongside this change in this PR?

Reply:
>  loop would run N times

loop will run N times but will read from channels N-1 times -  because one time will spent in `case <-ctx.Done()`
Anyway, fixed it